test/stdlib/Result.swift

```diff
@@ -0,0 +1,178 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import Swift
+
+let ResultTests = TestSuite("Result")
+
+fileprivate enum Err: Error, Equatable {
+  case err
+  case derr
+}
+
+fileprivate let string = "string"
+
+fileprivate extension Result {
+  var value: Value? {
+    switch self {
+    case let .value(value):
+      return value
+    case .error:
+      return nil
+    }
+    
+    var error: Error? {
+      switch self {
+      case .value:
+        return nil
+      case let .error(error):
+        return error
+      }
+    }
+  }
+}
+
+ResultTests.test("Construction") {
+  let result1: Result<String, Err> = .value(string)
+  let result2: Result<String, Err> = .error(.err)
+  let string1: String? = {
+    switch result1 {
+      case let .value(string): 
+        return string
+      case .error: 
+        expectUnreachable()
+        return nil
+    }
+  }()
+  let error: Err? = {
+    switch result2 {
+      case let .error(error): 
+        return error
+      case .value: 
+        expectUnreachable()
+        return nil
+    }
+  }()
+
+  expectEqual(string1, string)
+  expectEqual(error, .err)
+}
+
+ResultTests.test("Throwing Initialization and Unwrapping") {
+  func notThrowing() throws -> String {
+    return string
+  }
+
+  func throwing() throws -> String {
+    throw Err.err
+  }
+    
+  let result1 = Result { try throwing() }
+  let result2 = Result { try notThrowing() }
+  
+  // Kept getting enum case 'error' cannot be used as an instance member, so get value manually.
```

Comment:
What were you trying to write?

Reply:
`expectEqual(result1.error, Err.err)`

Reply:
Why would you expect the first to compile?  We removed that accessor.

Reply:
I should say, `expectEqual(result1.error as? Err, Err.err)`.

Reply:
Fixed in [7fb5905](https://github.com/apple/swift/pull/20958/commits/7fb5905f5f5b9076769508e35882bcd1380a755e).